openapi.yaml

```diff
@@ -1617,6 +1632,187 @@ paths:
           $ref: '#/components/responses/client_error_auth'
         5XX:
           $ref: '#/components/responses/server_error'
+  '/processes/{namespace}':
+    get:
+      summary: List all user-defined processes in a namespace
```

Comment:
Documentation of `/processes/{namespace}` and `/processes/{namespace}/{process_id}` talk about "user-defined processes in a "namespace", but listing "predefined" processes should also work, right?

e.g. `GET /processes/backend` returns the same as `GET /processes`
and `GET /process/backend/filter_temporal`  would return the metadata of `filter_temporal` process?

Reply:
I think the thinking was that there would be duplication as you've pointed out. You usually get all details from /processes for pre-defined processes so that this "extension" is mostly for user-defined processes (and was mostly copied from /process_graphs). So we can discuss whether we should remove the "user-defined" here. The backend namespace is somewhat different though as it is read-only...

Reply:
Yes, I'm fine with that. We can add additional wording that clarifies any specifics for user-defined processes.

Reply:
Hmm, looking into the OpenAPI file made me remember that user-defined processes and pre-defined processes have a different schema. So indeed the endpoints were only meant to support user-defined processes so far.

Reply:
So it seems we need to split up into two separate endpoints to make sure the schemas can be applied correctly:
- '/processes/{namespace}': user-defined
- '/processes/backend': predefined

Reply:
hmm that's unfortunate. So that means there are still quite some, possibly breaking, steps to take to reach the ultimate goal of "process unification"?

Reply:
Yes, could be. Still waiting for more feedback, not ideal if it's just us commenting here.
The first question of course is what "process unification" means? Right now it's just about API endpoints, but maybe we also want to go deeper into the required fields? The schemas itself are the same, except for required fields...

Reply:
> we need to split up into two separate endpoints to make sure the schemas can be applied correctly
> Or we don't allow exposing this endpoint as it's already exposed via /processes. 

Both options are equally bad for us, VITO, as we are already using a non-"backend" namespace containing predefined proprietary processes (without a "process_graph"), which would be invalid according to both of these options.

Reply:
Hmm... maybe we can add a discriminator to GET /processes/{namespace}. If it contains type: `user: true` in the response it applies the user schema to the processes array, otherwise the predefined schema. That would also help clients to know whether they can do non-GET requests on that namespace.

Reply:
I think part of the current problem with "predefined" vs "user-defined", is that we are lumping together a couple of process concepts, for example:

predefined | user defined
-----------|-----------
live in default namespace | live in "user" namespace
implemented "natively" by backend | implemented through openEO process graph
has no "process_graph" field in metadata | has a "process_graph" field in metadata
public (by default?) | private (by default?)
no public API to add/update/remove  |  created and managed though openEO endpoints
parameters and return values must be declared | parameters and return values are optional

By sticking to this binary division, each with own "schema", we probably make it hard for ourselves to create a clean API in the long term. 
For example, in VITO backend we already have processes that mix properties from both columns, e.g. private, natively implemented processes that live in a namespace that is neither default or per-user. Another example is defining predefined processes through a "process_graph", instead of "natively" (e.g. "ndvi" or "evi").

Reply:
Please remember that we are just moving things around here to stay compliant with the API v1.x line. We can't change a lot wrt the schemas for example as that usually is a breaking change. So until we go for API v2 we may need to go with some compromises.

Reply:
sure, I understand, I was just reflecting on the tension between the constraints of the v1.x API and what we are doing in VITO backend (or want to do) with custom namespaces